packages/content/types/content.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { Database } from './database';
+import type { QueryBuilder } from './query-builder'
+
+export type contentFunc = (...args: Array<string | Object>) => QueryBuilder;
+
+export interface IContentDocument extends Record<string, any> {
```

Comment:
Yes it would be nice to document this but it would be even better to extend this part later if we want more TypeScript support. This way we can cast the any as a different generic you want to have based on your input

Reply:
We could use `export interface IContentDocument<T = Record<string, any>> extends T {`

At fetch we could use `fetch<T>(): IContentDocument<T> | IContentDocument<T>[];`

Reply:
Unfortunately I do not get this solution to work. You can not extend your own generic. I think the best way to implement this is to keep it like an extension of a Record. 
If you want to call the fetch you can do the following:
```
import { IContentDocument } from '@nuxt/content/types/content'; 

interface YourOwnModel extends IContentDocument {
	yourkey: any;
}

const data = $content('markdown').fetch<YourOwnModel>();
```

Reply:
Or maybe change the fetch function?
```
fetch(): IContentDocument | IContentDocument[];
fetch<T>(): T & IContentDocument | T[] & IContentDocument[];
```

Reply:
I think this is better
```ts
fetch(): IContentDocument | IContentDocument[];
fetch<T>(): T & IContentDocument | (T & IContentDocument)[];
```